stew/base36.nim

```diff
@@ -0,0 +1,238 @@
+## Copyright (c) 2018 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+## This module implements BASE36 encoding and decoding procedures.
+## This module supports two variants of BASE36 encoding (Bitcoin and Flickr).
+
+type
+  Base36Status* {.pure.} = enum
+    Error,
+    Success,
+    Incorrect,
+    Overrun
+
+  Base36Alphabet* = object
+    decode*: array[128, int8]
+    encode*: array[36, uint8]
+
+  Base36* = object
+  Base36Uc* = object
+  Base36Lc* = object
+
+  Base36Types* = Base36 | Base36Uc | Base36Lc
+
+  Base36Error* = object of CatchableError
+    ## Base36 specific exception type
+
+proc newAlphabet36*(s: string): Base36Alphabet =
+  doAssert(len(s) == 36)
+  for i in 0..<len(s):
+    result.encode[i] = cast[uint8](s[i])
+  for i in 0..<len(result.decode):
+    result.decode[i] = -1
+  for i in 0..<len(result.encode):
+    result.decode[int(result.encode[i])] = int8(i)
+
+const
+  B36UcAlphabet* = newAlphabet36("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+  B36LcAlphabet* = newAlphabet36("0123456789abcdefghijklmnopqrstuvwxyz")
+
+proc encodedLength*(btype: typedesc[Base36Types], length: int): int =
```

Comment:
https://status-im.github.io/nim-style-guide/language.proc.html#functions-and-procedures-languageproc
> Prefer `func` - use `proc` when side effects cannot conveniently be avoided.

Likewise with the rest of the `proc`s here. If they can't all be `func`, something is strange.